lib/src/deprecation.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:collection/collection.dart';
+import 'package:pub_semver/pub_semver.dart';
+
+/// Represents a deprecated feature in the language.
+enum Deprecation {
+  /// Deprecation for passing a string to `call` instead of `get-function`.
+  callString('call-string',
+      deprecatedIn: '0.0.0',
+      description: 'Passing a string directly to meta.call().'),
+
+  /// Deprecation for `@elseif`.
+  elseIf('elseif', deprecatedIn: '1.3.2', description: '@elseif.'),
```

Comment:
Nit: I think the `i` should be lowercase here to match the at-rule.

Reply:
Done